lib/components/Label.js

```diff
@@ -0,0 +1,29 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+/*
+ * renders text on top of children components
+ */
+class Label extends PureComponent {
+  static get propTypes() {
+    return {
+      text: PropTypes.string,
+      children: PropTypes.oneOfType([
+        PropTypes.node,
+        PropTypes.arrayOf(PropTypes.node),
+      ]),
+    };
+  }
+
+  render() {
+    const { text, children } = this.props;
+    return [
+      <div key={0} style={{ width: '100%', textAlign: 'inherits' }}>
```

Comment:
For components that are intended to eventually be common components in bpanel-ui, I think it would be good to give them a className as well so that we can target with common stylings via the theming system.